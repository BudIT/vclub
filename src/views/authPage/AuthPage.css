Form
{
    display: flex;
    justify-content: center;
    flex-direction: column;
    margin-top: 15vh;
}
form::after,
form::before
{
    box-sizing: border-box;
}

.form_group
{
    display: flex;
    flex-direction: column;
    font-family: sans-serif;
    font-size: 16px;
    padding: 20px;
    width: 25vw;
    line-height: 1.75em;
    background: azure;
    border: 1px solid rgba(0, 0, 0, .3);
    border-radius: 3px;
    color: rgba(0, 0, 0, .8);
}

span
{
    text-align: center;
    font-size: 18px;
    margin-top: 20px;
    margin-bottom: -7vh;
}
input[type=text]
{
    width: 24vw;
    height: 5vh;
    border: 1px solid rgba(255, 133, 51, .5);
    border-radius: 3px;
    font-size: 16px;
    padding: 5px;
    color: rgb(255, 133, 51);
}

.validation-error,
.invalid-field
{
    font-size: 12px;
}

.input_group
{
    flex-direction: column;
    display: flex;
    margin: 20px 0  0 70px;
}

input:focus
{
    outline: none;
}

.btn
{
    border-radius: 5px 33px;
    cursor: pointer;
    width: 200px;
    height: 30px;
    padding: 3px;
    margin: 5vh auto;
    font: 18px sans-serif;
    background: rgba(0, 0, 0, .8);
    border: 0;
    color: rgb(255, 133, 51);
    text-transform: uppercase;
}

.btn:hover
{
    background: rgb(0, 0, 0);
}

.btn:focus
{
    outline: none;
}
