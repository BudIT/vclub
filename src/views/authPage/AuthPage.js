import React from 'react';
import { Field, reduxForm } from 'redux-form';

import compose from 'recompose/compose';
import withHandlers from 'recompose/withHandlers';
import setPropTypes from 'recompose/setPropTypes';

import { auth } from 'vclub/redux/club/auth';

import styles from './AuthPage.css';


const validate = values => {
  const errors = {};

  if (!values.username) {
    errors.username = 'Пожалуйста, введите Ваше имя !!!';
  } else if (values.username.length > 15) {
    errors.username = 'Имя должно быть 15 символов или меньше';
  }

  return errors;
};

// eslint-disable-next-line react/prop-types
const AuthInputField = ({ input, type, placeholder, className, meta: { touched, error } }) => (
  <div>
    <div>
      <input
        {...input}
        type={type}
        placeholder={placeholder}
        className={className}
        autoFocus
      />
      {touched
      && error
      && <span className={styles.errors}>{error}</span>}
    </div>
  </div>
);

const enhance = compose(
  setPropTypes({
    dispatch: React.PropTypes.func.isRequired,
  }),
  withHandlers({
    onSubmit: (props) => (data) => {
      const { dispatch } = props;
      const { username, master, remember } = data;

      dispatch(auth({ username, master }, remember));
    },
  }),
  reduxForm({
    form: 'auth',
    initialValues: {
      username: '',
      master: false,
      remember: true,
    },
    validate,
  }),
);

const AuthPage = (props) => {
  const { handleSubmit } = props;

  return (
    <section className={styles.auth}>
      <form className={styles.login} onSubmit={handleSubmit}>
        <fieldset className={styles.form_group}>
          <div>
            <div>
              <Field
                name="username"
                component={AuthInputField}
                type="text"
                placeholder="Имя..."
                className={styles.input_name}
                onFocus
              />
            </div>
          </div>
          <div className={styles.check_container}>
            <div className={styles.round_checkbox}>
              <Field
                name="master"
                id="master"
                component="input"
                type="checkbox"
              />
              <label htmlFor="master">
                <h3 className={styles.label_h3}>Ведущий</h3>
              </label>
            </div>
            <div className={styles.round_checkbox}>
              <Field
                name="remember"
                id="remember"
                component="input"
                type="checkbox"
              />
              <label htmlFor="remember">
                <h3 className={styles.label_h3}>Запомнить</h3>
              </label>
            </div>
          </div>
          <button
            className={styles.btn}
            type="submit"
          >
            Войти
          </button>
        </fieldset>
      </form>
    </section>
  );
};

AuthPage.propTypes = {
  handleSubmit: React.PropTypes.func.isRequired,
};

export default enhance(AuthPage);
