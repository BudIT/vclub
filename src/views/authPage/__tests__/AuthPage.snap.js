/* eslint import/no-extraneous-dependencies: ["error", {"devDependencies": true}] */
/* eslint-env jest */

import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';

import { AuthPage } from '../AuthPage';

test('<AuthPage /> component renders correctly', () => {
  // const store = createStore(() => ({}));
  const shallowRenderer = ReactTestUtils.createRenderer();

  shallowRenderer.render(
    <AuthPage handleSubmit={jest.fn()} />
  );
  expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
});
