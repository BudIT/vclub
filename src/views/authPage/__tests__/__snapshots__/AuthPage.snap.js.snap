exports[`test <AuthPage /> component renders correctly 1`] = `
<section
  className="auth">
  <form
    className="login"
    onSubmit={[Function]}>
    <fieldset
      className="form_group">
      <div>
        <div>
          <function Field(props, context) {
                _classCallCheck(this, Field);
          
                var _this = _possibleConstructorReturn(this, (Field.__proto__ || Object.getPrototypeOf(Field)).call(this, props, context));
          
                if (!context._reduxForm) {
                  throw new Error('Field must be inside a component decorated with reduxForm()');
                }
          
                _this.normalize = _this.normalize.bind(_this);
                return _this;
              }
            className="input_name"
            component={[Function]}
            name="username"
            onFocus={true}
            placeholder="Имя..."
            type="text" />
        </div>
      </div>
      <div
        className="check_container">
        <div
          className="round_checkbox">
          <function Field(props, context) {
                _classCallCheck(this, Field);
          
                var _this = _possibleConstructorReturn(this, (Field.__proto__ || Object.getPrototypeOf(Field)).call(this, props, context));
          
                if (!context._reduxForm) {
                  throw new Error('Field must be inside a component decorated with reduxForm()');
                }
          
                _this.normalize = _this.normalize.bind(_this);
                return _this;
              }
            component="input"
            id="master"
            name="master"
            type="checkbox" />
          <label
            htmlFor="master">
            <h3
              className="label_h3">
              Ведущий
            </h3>
          </label>
        </div>
        <div
          className="round_checkbox">
          <function Field(props, context) {
                _classCallCheck(this, Field);
          
                var _this = _possibleConstructorReturn(this, (Field.__proto__ || Object.getPrototypeOf(Field)).call(this, props, context));
          
                if (!context._reduxForm) {
                  throw new Error('Field must be inside a component decorated with reduxForm()');
                }
          
                _this.normalize = _this.normalize.bind(_this);
                return _this;
              }
            component="input"
            id="remember"
            name="remember"
            type="checkbox" />
          <label
            htmlFor="remember">
            <h3
              className="label_h3">
              Запомнить
            </h3>
          </label>
        </div>
      </div>
      <button
        className="btn"
        type="submit">
        Войти
      </button>
    </fieldset>
  </form>
</section>
`;
